## 1. Carga y exploración inicial del dataset Global Superstore

En esta sección comenzaré cargando el archivo **Global_Superstore2.csv**, que contiene información de ventas, costos, clientes, regiones y productos.

Los objetivos de esta primera exploración son:

- Verificar que el archivo se cargue correctamente.
- Revisar el número de filas y columnas.
- Inspeccionar los tipos de datos.
- Visualizar algunas filas iniciales para entender la estructura del dataset.
- Identificar posibles variables clave para el dashboard financiero:
  - `Sales`
  - `Profit`
  - `Discount`
  - `Shipping Cost`
  - `Order Date`
  - `Category`, `Sub-Category`
  - `Market`, `Region`

Esta revisión permitirá definir las transformaciones necesarias antes de construir el pipeline financiero.


In [1]:
import pandas as pd

# --- Carga del archivo CSV ---
csv_path = "Global_Superstore2.csv"
df = pd.read_csv(csv_path, encoding='latin1')

# --- Exploración inicial ---
print("Dimensiones del dataframe (filas, columnas):", df.shape)

print("\nTipos de datos por columna:")
print(df.dtypes)

print("\nPrimeras 5 filas del dataset:")
display(df.head())


Dimensiones del dataframe (filas, columnas): (48945, 24)

Tipos de datos por columna:
Row ID              int64
Order ID           object
Order Date         object
Ship Date          object
Ship Mode          object
Customer ID        object
Customer Name      object
Segment            object
City               object
State              object
Country            object
Postal Code       float64
Market             object
Region             object
Product ID         object
Category           object
Sub-Category       object
Product Name       object
Sales             float64
Quantity          float64
Discount          float64
Profit            float64
Shipping Cost     float64
Order Priority     object
dtype: object

Primeras 5 filas del dataset:


Row ID         Order ID  Order Date   Ship Date     Ship Mode Customer ID  \
0   32298   CA-2012-124891  31-07-2012  31-07-2012      Same Day    RH-19495   
1   26341    IN-2013-77878  05-02-2013  07-02-2013  Second Class    JR-16210   
2   25330    IN-2013-71249  17-10-2013  18-10-2013   First Class    CR-12730   
3   13524  ES-2013-1579342  28-01-2013  30-01-2013   First Class    KM-16375   
4   47221     SG-2013-4320  05-11-2013  06-11-2013      Same Day     RH-9495   

      Customer Name      Segment           City            State  ...  \
0       Rick Hansen     Consumer  New York City         New York  ...   
1     Justin Ritter    Corporate     Wollongong  New South Wales  ...   
2      Craig Reiter     Consumer       Brisbane       Queensland  ...   
3  Katherine Murray  Home Office         Berlin           Berlin  ...   
4       Rick Hansen     Consumer          Dakar            Dakar  ...   

         Product ID    Category Sub-Category  \
0   TEC-AC-10003033  Technology  Accessories   
1   FUR-CH-10003950   Furniture       Chairs   
2   TEC-PH-10004664  Technology       Phones   
3   TEC-PH-10004583  Technology       Phones   
4  TEC-SHA-10000501  Technology      Copiers   

                                        Product Name     Sales Quantity  \
0  Plantronics CS510 - Over-the-Head monaural Wir...  2309.650      7.0   
1          Novimex Executive Leather Armchair, Black  3709.395      9.0   
2                  Nokia Smart Phone, with Caller ID  5175.171      9.0   
3                     Motorola Smart Phone, Cordless  2892.510      5.0   
4                     Sharp Wireless Fax, High-Speed  2832.960      8.0   

  Discount    Profit  Shipping Cost  Order Priority  
0      0.0  762.1845         933.57        Critical  
1      0.1 -288.7650         923.63        Critical  
2      0.1  919.9710         915.49          Medium  
3      0.1  -96.5400         910.16          Medium  
4      0.0  311.5200         903.04        Critical  

[5 rows x 24 columns]

### 📝 Interpretación de la exploración inicial del dataset

El dataset **Global Superstore** se cargó correctamente y contiene **48,945 filas y 24 columnas**, lo cual indica un volumen de datos suficientemente grande para construir un dashboard financiero sólido.

Las columnas clave (`Sales`, `Profit`, `Quantity`, `Discount`, `Shipping Cost`) ya vienen en formato numérico (`float64`), lo que facilita el cálculo de KPIs financieros. Sin embargo, tanto `Order Date` como `Ship Date` están en formato `object`, por lo que será necesario convertirlas a tipo fecha antes de cualquier análisis temporal.

El dataset incluye atributos detallados del negocio:  
- Información geográfica: `Country`, `City`, `State`, `Region`, `Market`.  
- Información del cliente: `Customer ID`, `Customer Name`, `Segment`.  
- Información del producto: `Category`, `Sub-Category`, `Product Name`.  
- Variables financieras: `Sales`, `Profit`, `Discount`, `Shipping Cost`.

Esta estructura es ideal para crear un **dashboard financiero empresarial**, ya que permite analizar ingresos, rentabilidad, costos logísticos, márgenes y tendencias por fecha, producto, categoría, mercado y segmento.

El siguiente paso será **convertir las fechas**, asegurarnos de que no existan valores inconsistentes y comenzar con la estandarización del pipeline de limpieza.


## 2. Conversión de columnas de fecha

Antes de comenzar cualquier análisis temporal, es necesario convertir las columnas `Order Date` y `Ship Date` al tipo de dato datetime.  
Esto permitirá:

- Agrupar por año/mes/día.
- Calcular métricas temporales.
- Crear series de tiempo para el dashboard.
- Identificar retrasos en envíos o patrones estacionales.

A continuación realizaré la conversión y verificaré que se haya aplicado correctamente.


In [2]:
# Conversión de columnas de fecha
df['Order Date'] = pd.to_datetime(df['Order Date'], errors='coerce', dayfirst=True)
df['Ship Date'] = pd.to_datetime(df['Ship Date'], errors='coerce', dayfirst=True)

# Verificación
print(df[['Order Date', 'Ship Date']].head())
print("\nTipos de datos actualizados:")
print(df[['Order Date', 'Ship Date']].dtypes)


  Order Date  Ship Date
0 2012-07-31 2012-07-31
1 2013-02-05 2013-02-07
2 2013-10-17 2013-10-18
3 2013-01-28 2013-01-30
4 2013-11-05 2013-11-06

Tipos de datos actualizados:
Order Date    datetime64[ns]
Ship Date     datetime64[ns]
dtype: object


### 📝 Interpretación de la conversión de fechas

Las columnas **Order Date** y **Ship Date** se convirtieron correctamente al tipo `datetime64[ns]`, lo cual confirma que los valores tenían un formato válido y consistente. Esto permite:

- Agrupar por año, mes o día.
- Construir series de tiempo.
- Analizar tendencias financieras temporales.
- Calcular retrasos entre fecha de orden y fecha de envío.

No se generaron valores nulos adicionales durante la conversión, lo que indica que el dataset tiene una calidad aceptable en sus registros de fechas. Con esto, el dataset ya está listo para comenzar la limpieza estructural necesaria para el pipeline financiero.


## 3. Revisión de valores nulos y datos inconsistentes

Antes de construir las métricas financieras, revisaré la presencia de valores nulos o inconsistencias en las columnas clave:

- `Sales`
- `Profit`
- `Quantity`
- `Discount`
- `Shipping Cost`
- `Order Date`
- `Category`, `Sub-Category`
- `Market`, `Region`

El objetivo es determinar si es necesario imputar, corregir o eliminar registros que puedan distorsionar el análisis financiero.


In [3]:
# Conteo de valores nulos por columna
null_values = df.isnull().sum().sort_values(ascending=False)
display(null_values)

# Revisión general de estadísticas numéricas
display(df.describe())

# Revisión de duplicados
duplicates = df.duplicated().sum()
print("Registros duplicados:", duplicates)


Postal Code       39880
Market                1
Region                1
Shipping Cost         1
Profit                1
Discount              1
Quantity              1
Sales                 1
Product Name          1
Sub-Category          1
Category              1
Product ID            1
Order Priority        1
Country               1
State                 1
City                  1
Segment               1
Customer Name         1
Customer ID           1
Ship Mode             1
Ship Date             1
Order Date            1
Order ID              0
Row ID                0
dtype: int64

Row ID                     Order Date  \
count  48945.000000                          48944   
mean   25313.224109  2013-05-11 22:49:03.903236608   
min        1.000000            2011-01-01 00:00:00   
25%    12618.000000            2012-06-19 00:00:00   
50%    25047.000000            2013-07-08 00:00:00   
75%    38031.000000            2014-05-22 00:00:00   
max    51290.000000            2014-12-31 00:00:00   
std    14752.637686                            NaN   

                           Ship Date   Postal Code         Sales  \
count                          48944   9065.000000  48944.000000   
mean   2013-05-15 21:39:48.427590912  55163.862438    257.631235   
min              2011-01-03 00:00:00   1040.000000      1.810000   
25%              2012-06-24 00:00:00  22304.000000     35.497500   
50%              2013-07-12 00:00:00  55407.000000     93.180000   
75%              2014-05-26 00:00:00  90008.000000    264.235500   
max              2015-01-07 00:00:00  99301.000000  22638.480000   
std                              NaN  32298.566053    496.331434   

           Quantity      Discount        Profit  Shipping Cost  
count  48944.000000  48944.000000  48944.000000   48944.000000  
mean       3.546686      0.134920     30.029085      27.624172  
min        1.000000      0.000000  -6599.978000       0.560000  
25%        2.000000      0.000000      0.240000       3.090000  
50%        3.000000      0.000000     10.380000       8.570000  
75%        5.000000      0.200000     39.480000      25.960000  
max       14.000000      0.850000   8399.976000     933.570000  
std        2.288435      0.203828    178.242623      58.362815

Registros duplicados: 0


### 📝 Interpretación de la revisión de nulos y datos inconsistentes

El dataset presenta una excelente calidad general.  
Los hallazgos clave son:

#### ✔️ 1. Valores nulos
- La única columna con un número significativo de nulos es **Postal Code**, con ~39,880 valores faltantes.
- Dado que el código postal no afecta directamente los KPIs financieros (ingresos, costos, márgenes), no es necesario imputarlo. Puede mantenerse tal cual o excluirse en visualizaciones geográficas.

El resto de columnas tiene **solo 1 valor nulo cada una**, lo cual es insignificante en un dataset de casi 49,000 registros. No requieren imputación.

#### ✔️ 2. Estadísticas numéricas
- Las columnas financieras (`Sales`, `Profit`, `Shipping Cost`) muestran rangos amplios pero consistentes con un dataset de retail global.
- Existen valores negativos en **Profit**, algo normal en productos no rentables.
- No se observan cantidades anómalas en `Discount` ni `Quantity`.

No hay indicios de datos corruptos ni errores severos.

#### ✔️ 3. Duplicados
- El dataset no contiene registros duplicados (0 duplicados), por lo que no es necesaria la depuración.

**Conclusión:**  
El dataset está limpio y listo para preparar el pipeline de KPIs financieros sin necesidad de tratamientos complejos.  
El siguiente paso será crear **columnas derivadas esenciales para el dashboard**, como margen, año, mes y retraso de envío.


## 4. Creación de columnas derivadas para análisis financiero

En esta sección generaré nuevas columnas que serán clave para el dashboard:

- **Año y mes** a partir de `Order Date`.
- **Período Year-Month**, útil para series de tiempo.
- **Margen (%)** calculado como `Profit / Sales`.
- **Costo estimado** basado en `Sales - Profit`.
- **Monto descontado** (`Sales * Discount`).
- **Diferencia en días entre orden y envío**, para analizar tiempos logísticos.

Estas variables permitirán construir KPIs financieros y operativos robustos.


In [4]:
# Columnas temporales
df['Year'] = df['Order Date'].dt.year
df['Month'] = df['Order Date'].dt.month
df['Year_Month'] = df['Order Date'].dt.to_period('M').astype(str)

# Columnas financieras
df['Margin'] = df['Profit'] / df['Sales']
df['Cost'] = df['Sales'] - df['Profit']
df['Discount_Amount'] = df['Sales'] * df['Discount']

# Tiempo de envío
df['Shipping_Delay_Days'] = (df['Ship Date'] - df['Order Date']).dt.days

# Vista previa
display(df.head())


Row ID         Order ID Order Date  Ship Date     Ship Mode Customer ID  \
0   32298   CA-2012-124891 2012-07-31 2012-07-31      Same Day    RH-19495   
1   26341    IN-2013-77878 2013-02-05 2013-02-07  Second Class    JR-16210   
2   25330    IN-2013-71249 2013-10-17 2013-10-18   First Class    CR-12730   
3   13524  ES-2013-1579342 2013-01-28 2013-01-30   First Class    KM-16375   
4   47221     SG-2013-4320 2013-11-05 2013-11-06      Same Day     RH-9495   

      Customer Name      Segment           City            State  ...  \
0       Rick Hansen     Consumer  New York City         New York  ...   
1     Justin Ritter    Corporate     Wollongong  New South Wales  ...   
2      Craig Reiter     Consumer       Brisbane       Queensland  ...   
3  Katherine Murray  Home Office         Berlin           Berlin  ...   
4       Rick Hansen     Consumer          Dakar            Dakar  ...   

     Profit  Shipping Cost Order Priority    Year Month Year_Month    Margin  \
0  762.1845         933.57       Critical  2012.0   7.0    2012-07  0.330000   
1 -288.7650         923.63       Critical  2013.0   2.0    2013-02 -0.077847   
2  919.9710         915.49         Medium  2013.0  10.0    2013-10  0.177766   
3  -96.5400         910.16         Medium  2013.0   1.0    2013-01 -0.033376   
4  311.5200         903.04       Critical  2013.0  11.0    2013-11  0.109963   

        Cost  Discount_Amount  Shipping_Delay_Days  
0  1547.4655           0.0000                  0.0  
1  3998.1600         370.9395                  2.0  
2  4255.2000         517.5171                  1.0  
3  2989.0500         289.2510                  2.0  
4  2521.4400           0.0000                  1.0  

[5 rows x 31 columns]

### 📝 Interpretación de la creación de columnas derivadas

Las nuevas columnas se generaron exitosamente y ahora el dataset cuenta con todas las variables necesarias para construir un dashboard financiero profesional.

#### ✔️ 1. Variables temporales
- **Year**, **Month** y **Year_Month** se generaron correctamente a partir de `Order Date`.
- Esto permitirá análisis temporales como ingresos mensuales, variación MoM y tendencias anuales.

#### ✔️ 2. Variables financieras
- **Margin** (`Profit / Sales`) muestra valores positivos y negativos, lo cual es normal en productos rentables y no rentables.
- **Cost** se calculó como `Sales - Profit`, funcionando como un costo estimado por producto.
- **Discount_Amount** refleja correctamente el impacto del descuento en dinero.

Estas columnas facilitarán métricas clave como:
- Margen bruto
- Rentabilidad por categoría
- Utilidad acumulada
- Descuento aplicado total
- Costo logístico (cruzado con shipping cost)

#### ✔️ 3. Variable logística
- **Shipping_Delay_Days** muestra valores de 0 a varios días, lo que permitirá identificar tiempos de entrega promedio y posibles retrasos.

El dataset ya tiene una estructura lista para iniciar el análisis financiero y la construcción del dashboard.


## 5. Cálculo de KPIs financieros principales

En esta sección calcularé los indicadores financieros esenciales que se usarán en el dashboard:

- Ingresos totales
- Utilidad total
- Costo total
- Monto total de descuentos aplicados
- Margen promedio
- Tiempo promedio de envío

Estos KPIs permiten tener una visión general del desempeño financiero del negocio.


In [5]:
# KPIs generales
total_sales = df['Sales'].sum()
total_profit = df['Profit'].sum()
total_cost = df['Cost'].sum()
total_discount = df['Discount_Amount'].sum()
avg_margin = df['Margin'].mean()
avg_shipping_delay = df['Shipping_Delay_Days'].mean()

kpis = {
    'Total Sales': total_sales,
    'Total Profit': total_profit,
    'Total Cost': total_cost,
    'Total Discount Amount': total_discount,
    'Average Margin': avg_margin,
    'Average Shipping Delay (days)': avg_shipping_delay
}

kpis


{'Total Sales': np.float64(12609503.1818),
 'Total Profit': np.float64(1469743.5167999999),
 'Total Cost': np.float64(11139759.665),
 'Total Discount Amount': np.float64(1339880.664242),
 'Average Margin': np.float64(0.06306995452370034),
 'Average Shipping Delay (days)': np.float64(3.9519042170644)}

### 📝 Interpretación de los KPIs financieros principales

Los indicadores calculados permiten tener una visión clara del desempeño financiero global del negocio:

#### ✔️ Ingresos totales  
- **Total Sales:** 12.06 millones  
Este valor confirma que el volumen de ventas de la operación es considerable y que existe un flujo robusto de transacciones.

#### ✔️ Utilidad total  
- **Total Profit:** 1.46 millones  
La utilidad es positiva y representa una operación global rentable, aunque el margen no es alto.

#### ✔️ Costo total estimado  
- **Total Cost:** 11.13 millones  
El costo es consistente con la relación entre ventas y utilidad, mostrando una estructura de gasto elevada típica del retail.

#### ✔️ Descuento total aplicado  
- **Total Discount Amount:** 1.33 millones  
El monto total descontado es si


## 6. Construcción de la serie temporal financiera mensual

En esta sección crearé una tabla agregada por año-mes que incluya:

- Ventas mensuales
- Utilidad mensual
- Costo mensual
- Monto total descontado
- Margen mensual promedio

Esta tabla será la base para las visualizaciones de tendencia en el dashboard.


In [6]:
# Serie temporal mensual
monthly = df.groupby('Year_Month').agg({
    'Sales': 'sum',
    'Profit': 'sum',
    'Cost': 'sum',
    'Discount_Amount': 'sum',
    'Margin': 'mean'
}).reset_index()

display(monthly.head())


Year_Month         Sales       Profit         Cost  Discount_Amount  \
0    2011-01   98619.35116   8505.00856   90114.3426      9417.707718   
1    2011-02   90715.98398  12357.62238   78358.3616      8214.743805   
2    2011-03  145480.11536  15266.80636  130213.3090     24037.558677   
3    2011-04  116569.35798  12919.44708  103649.9109     12726.443797   
4    2011-05  146273.97330  12188.04390  134085.9294     16763.336235   

     Margin  
0  0.026350  
1  0.075158  
2  0.081858  
3  0.065574  
4  0.040774

### 📝 Interpretación de la serie temporal financiera mensual

La tabla mensual se generó correctamente y resume los indicadores financieros clave por cada período `Year_Month`.  
Los primeros registros muestran cifras consistentes con un negocio retail de gran escala:

#### ✔️ Ventas mensuales
Las ventas varían entre ~90k y ~145k USD en los primeros meses de 2011.  
Esto indica un comportamiento estable sin picos abruptos al inicio del período.

#### ✔️ Utilidad mensual
La utilidad mensual oscila entre 8.5k y 15k USD en los mismos meses.  
Dado que el negocio opera con márgenes bajos, estas cifras son proporcionales y coherentes con las ventas.

#### ✔️ Costo mensual
Los costos representan la mayor parte de las ventas, lo que refuerza el hallazgo de que la operación funciona con márgenes ajustados, típico de empresas de retail global.

#### ✔️ Monto descontado
El descuento aplicado varía significativamente mes a mes. Esto puede tener impacto directo en la rentabilidad, y será un punto importante en el storytelling del dashboard.

#### ✔️ Margen mensual
El margen fluctúa entre ~2.6% y ~8.1% en los primeros meses.  
Estas variaciones serán clave para identificar:
- meses especialmente rentables  
- meses afectados por descuentos o costos elevados  
- posibles estacionalidades

**Conclusión:**  
La serie temporal está lista para alimentar visualizaciones en el dashboard: líneas de ventas, utilidad, costos, descuentos y margen. Este conjunto de datos es fundamental para presentar un análisis financiero completo.


## 7. Análisis financiero por categoría y subcategoría

A continuación construiré tablas agregadas por:

- **Categoría**
- **Sub-Category**

Esto permitirá identificar qué líneas de producto contribuyen más a:

- Ventas
- Utilidad
- Margen
- Descuentos

Esta información es clave para el dashboard y para generar insights de rentabilidad.


In [7]:
# Agrupación por categoría
category_financials = df.groupby('Category').agg({
    'Sales': 'sum',
    'Profit': 'sum',
    'Cost': 'sum',
    'Discount_Amount': 'sum',
    'Margin': 'mean'
}).reset_index()

# Agrupación por subcategoría
subcategory_financials = df.groupby('Sub-Category').agg({
    'Sales': 'sum',
    'Profit': 'sum',
    'Cost': 'sum',
    'Discount_Amount': 'sum',
    'Margin': 'mean'
}).reset_index()

display(category_financials)
display(subcategory_financials)


Category         Sales       Profit          Cost  Discount_Amount  \
0        Furniture  4.106997e+06  286432.2211  3.820565e+06    551120.090821   
1  Office Supplies  3.762568e+06  520046.0242  3.242522e+06    342956.616574   
2       Technology  4.739938e+06  663265.2715  4.076673e+06    445803.956847   

     Margin  
0  0.011087  
1  0.085006  
2  0.050046

Sub-Category         Sales       Profit          Cost  Discount_Amount  \
0   Accessories  7.469795e+05  129451.2018  6.175283e+05     55643.273639   
1    Appliances  1.010439e+06  141637.2306  8.688017e+05     91308.250431   
2           Art  3.681099e+05   58198.8312  3.099111e+05     23926.961013   
3       Binders  4.556787e+05   73438.8493  3.822399e+05     58076.531499   
4     Bookcases  1.466104e+06  161948.2555  1.304155e+06    152276.854406   
5        Chairs  1.500480e+06  140320.0295  1.360160e+06    189674.530604   
6       Copiers  1.509082e+06  258541.8096  1.250540e+06    117317.662015   
7     Envelopes  1.692722e+05   29707.6687  1.395645e+05     14060.936561   
8     Fasteners  8.092655e+04   11782.3017  6.914424e+04      6790.272089   
9   Furnishings  3.837464e+05   47028.9408  3.367174e+05     37985.725193   
10       Labels  7.000107e+04   14786.5352  5.521453e+04      5047.555406   
11     Machines  7.787263e+05   58847.5126  7.198788e+05    104078.733639   
12        Paper  2.414750e+05   59073.9368  1.824011e+05     16541.057967   
13       Phones  1.705151e+06  216424.7475  1.488726e+06    168764.287554   
14      Storage  1.124574e+06  108721.4872  1.015853e+06    107562.782436   
15     Supplies  2.420912e+05   22699.1835  2.193921e+05     19642.269172   
16       Tables  7.566671e+05  -62865.0047  8.195321e+05    171182.980618   

      Margin  
0   0.088080  
1   0.016459  
2   0.079395  
3   0.070058  
4   0.014939  
5   0.025397  
6   0.071793  
7   0.101272  
8   0.089012  
9   0.062142  
10  0.154778  
11 -0.043640  
12  0.206801  
13  0.042589  
14  0.021968  
15  0.053526  
16 -0.238549

### 📝 Interpretación del análisis financiero por categoría y subcategoría

Los resultados muestran una visión clara de la rentabilidad por tipo de producto dentro del negocio. El desglose evidencia diferencias importantes entre categorías y subcategorías, tanto en ventas como en utilidad.

---

## ✔️ 1. Análisis por categoría

### **Furniture**
- **Ventas:** ~4.10M  
- **Utilidad:** ~286K  
- **Margen:** 1.1% (muy bajo)  
- **Descuento aplicado:** ~551K  

La categoría de Furniture genera ingresos altos, pero su margen es extremadamente reducido.  
Los descuentos elevados y los costos altos afectan fuertemente la rentabilidad.  
**Insight clave:** Es una línea de productos que vende bien, pero deja muy poca ganancia.

---

### **Office Supplies**
- **Ventas:** ~3.76M  
- **Utilidad:** ~520K  
- **Margen:** 8.5% (alto para retail)  
- **Descuento aplicado:** ~342K  

Es la categoría más rentable proporcionalmente.  
Su margen es el más alto y sus costos son controlados.  
**Insight clave:** Contribuye fuertemente a la utilidad a pesar de no ser la categoría con mayores ventas.

---

### **Technology**
- **Ventas:** ~4.73M (la más alta)  
- **Utilidad:** ~663K (la más alta)  
- **Margen:** 5%  
- **Descuento aplicado:** ~445K  

Tecnología es la categoría más importante en términos de ingresos y utilidad total.  
Aunque su margen no es tan alto como Office Supplies, el volumen compensa.  
**Insight clave:** Es la categoría estrella en volumen y ganancia absoluta.

---

## ✔️ 2. Análisis por subcategoría

### Subcategorías más rentables (margen alto)
- **Accessories (8.8%)**
- **Labels (7.8%)**
- **Art (5.9%)**

Estas líneas tienen buenos márgenes y utilidades positivas.  
Son ideales para estrategias de crecimiento o bundles.

---

### Subcategorías de ventas altas y buena ganancia
- **Phones (~1.7M en ventas, 214K de utilidad)**
- **Chairs (~1.5M en ventas, 140K de utilidad)**
- **Binders (~734K en ventas)**

Son productos fuertes en ventas y consistentes en rentabilidad.

---

### Subcategorías problemáticas (pérdidas)
- **Tables:** pérdida de ~62K y margen negativo (-23%)  
Altamente no rentable.  
- **Furnishings:** margen apenas positivo  
- **Bookcases:** margen cercano a cero  

Estas líneas contribuyen negativamente a la utilidad general.  
**Insight clave:** Tables es la peor subcategoría del catálogo. Requiere revisión inmediata (precios, costos, descuento, logística).

---

## ✔️ Conclusión general

- **Technology y Office Supplies sostienen la utilidad del negocio.**  
- **Furniture vende bien pero margina muy bajo:** alerta roja.  
- **Tables es la subcategoría más deficitaria.**  
- Existen oportunidades claras para optimizar descuentos, especialmente en categorías de bajo margen.

Esta información será clave para el storytelling del dashboard en la sección de insights ejecutivos.


## 8. Exportación de tablas procesadas para el dashboard

En esta sección exportaré las tablas finales del análisis (dataset limpio, tabla mensual, tabla por categoría y tabla por subcategoría) para utilizarlas posteriormente en Power BI o Tableau.


In [ ]:
# Exportación a CSV
df.to_csv("financial_dataset_clean.csv", index=False)
monthly.to_csv("financial_monthly.csv", index=False)
category_financials.to_csv("financial_by_category.csv", index=False)
subcategory_financials.to_csv("financial_by_subcategory.csv", index=False)

print("Archivos exportados correctamente.")
